pytorch_forecasting/data/timeseries/_timeseries.py

```diff
@@ -1439,16 +1439,21 @@ def _to_tensor(cols, long=True) -> torch.Tensor:
             * float columns are converted to torch.float
             * integer columns are converted to torch.int64 or torch.long,
               depending on the long argument
+            * mixed columns are converted to their commonest type.
+            * if real argument is True, force the conversion to torch.float
             """
             if not isinstance(cols, list) and cols not in data.columns:
                 return None
             if isinstance(cols, list) and len(cols) == 0:
                 dtypekind = "f"
             elif isinstance(cols, list):  # and len(cols) > 0
-                dtypekind = data.dtypes[cols[0]].kind
+                # dtypekind = data.dtypes[cols[0]].kind
+                dtypekind = data[cols].to_numpy().dtype.kind
```

Comment:
why are we doing this? It feels wasteful to do the conversion for the entire data container, instead of looking up the dtype.